File: backend/models/customFieldModel.js
import mongoose from "mongoose";

const customFieldSchema = new mongoose.Schema(
  {
    companyId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Company",
      required: true,
    },
    name: {
      type: String,
      required: true,
      trim: true,
    },
    type: {
      type: String,
      enum: ["text", "number", "date", "dropdown", "checkbox", "textarea"],
      default: "text",
    },
    options: {
      type: [String], // Only required if type === 'dropdown'
      default: [],
    },
    placeholder: {
      type: String,
      default: "",
    },
    isRequired: {
      type: Boolean,
      default: false,
    },
  },
  { timestamps: true }
);

const CustomField = mongoose.model("CustomField", customFieldSchema);
export default CustomField;

File: frontend/lib/models/custom_field_model.dart
class CustomField {
  final String id;
  final String name;
  final String type; // 'text', 'number', 'date', 'dropdown', 'checkbox'
  final bool isRequired;
  final String placeholder; // Maps to 'description' in UI
  final List<String> options;

  CustomField({
    required this.id,
    required this.name,
    required this.type,
    required this.isRequired,
    required this.placeholder,
    required this.options,
  });

  factory CustomField.fromJson(Map<String, dynamic> json) {
    return CustomField(
      id: json['_id'] ?? json['id'] ?? '',
      name: json['name'] ?? '',
      type: json['type'] ?? 'text',
      isRequired: json['isRequired'] ?? false,
      placeholder: json['placeholder'] ?? '',
      options: (json['options'] as List<dynamic>?)
              ?.map((e) => e.toString())
              .toList() ??
          [],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'type': type,
      'isRequired': isRequired,
      'placeholder': placeholder,
      'options': options,
    };
  }
}
